src/modules/BookModals/presentation/components/AddBook.tsx

```
@@ -81,6 +81,47 @@ export function AddBookModal({ open, onOpenChange }: AddBookModalProps) {
+        if (message) return message;
+      }
+    }
+
+    return undefined;
+  };
+
+  const onInvalid: SubmitErrorHandler<FormValues> = (errors) => {
+    const message =
+      extractErrorMessage(errors) ||
+      "Please resolve the highlighted fields before submitting.";
+    useToast("error", message);
```

Biome (2.1.2)

[error] This hook is being called from a nested function, but all hooks must be called unconditionally from the top-level component.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)